app/api/prediction/route.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  getPredictionByMarketAddress,
+  savePrediction,
+} from "../../../lib/db/queries";
+import { generateText } from "ai";
+import { groq } from "@ai-sdk/groq";
+import { isAddress } from "viem";
+import { getMarket } from "../../../queries/omen/markets";
+import { NextResponse } from "next/server";
+import { generateSystemPrompt, jsonPrompt } from "../util";
+
+export async function GET(request: Request) {
+  const { searchParams } = new URL(request.url);
+  const marketId = searchParams.get("market_id");
+
+  const address =
+    marketId && isAddress(marketId) ? marketId.toLowerCase() : null;
+
+  if (!address) {
+    return new Response("No address provided", { status: 400 });
+  }
+
+  const prediction = await getPredictionByMarketAddress({
+    marketAddress: address,
+  });
+
+  return NextResponse.json({
+    message: {
+      response: prediction.content,
+    },
+    role: "assistant",
+  });
+}
+
+export async function POST(request: Request) {
+  const { marketId }: { marketId: string } = await request.json();
+
+  const address = isAddress(marketId) ? marketId.toLowerCase() : null;
+
+  if (!address) {
+    return new Response("No address provided", { status: 400 });
+  }
```

Comment:
this is just taste, but would get rid of all `{}`

I think it looks cleaner like this

```suggestion
  if (!address) return new Response("No address provided", { status: 400 });
```

Reply:
Would do the same for all of this if conditions

Reply:
I think in this case it's easier to perceive that's a condition, without them they look to close and gets a bit messy. All of them without braces looks like a normal piece of code, when it's better to distinguish if it's a condition. Basically you know where the if begins and ends.

Reply:
I'm okay with this, it's just a preference.